core/src/main/java/ai/timefold/solver/core/preview/api/variable/provided/ShadowVariableSession.java

```diff
@@ -0,0 +1,13 @@
+package ai.timefold.solver.core.preview.api.variable.provided;
+
+public interface ShadowVariableSession {
```

Comment:
My understanding of this API is still limited, but do we actually need this public API anywhere in the solver? IMO this is internal in core, and only exposed in the tests; therefore I'd make this public in the tests, not in the solver itself.

Reply:
It can be moved; it is here for convenience when testing the prototype. (core internally depends on the default implementation which has before/after methods).